modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/LZ4CompressionFactory.java

```diff
@@ -0,0 +1,216 @@
+ */
+public class LZ4CompressionFactory {
+
+    /**
+     * Whether message content compression from the server side is enabled
+     */
+    static final boolean isCompressionEnabled = (boolean) AndesConfigurationManager.readValue
+            (AndesConfiguration.PERFORMANCE_TUNING_ALLOW_COMPRESSION);
+
+    /**
+     * Maximum content size to check before compress message content, in bytes. Messages with
+     * content less than this size will not compress.
+     */
+    static final int contentCompressionThreshold = (int) AndesConfigurationManager.readValue
+            (AndesConfiguration.PERFORMANCE_TUNING_CONTENT_COMPRESSION_THRESHOLD);
+
+    /**
+     * Keep a reference to lz4 instance
+     */
+    private static final LZ4Factory factory = LZ4Factory.fastestInstance();
+
+    /**
+     * Keep a reference to lz4 fast compressor to compress data
+     */
+    private static final LZ4Compressor compressor = factory.fastCompressor();
+
+    /**
+     * Keep a reference to lz4 fast decompressor to decompress data
+     */
+    private static final LZ4FastDecompressor decompressor = factory.fastDecompressor();
+
+    /**
+     * Compress the content of the message, using andes message part list
+     *
+     * @param partList              Original content chunk list
+     * @param originalContentLength Total content length of the original message content
+     * @return Compressed message content as an AndesMessagePart
+     */
+    public AndesMessagePart getCompressedMessage(List<AndesMessagePart> partList, int originalContentLength) {
+
+        byte[] messageData = getByteArrayFromPartList(partList, originalContentLength);
+
+        // Compress message content
+        int maxCompressedLength = compressor.maxCompressedLength(originalContentLength);
+        byte[] compressed = new byte[maxCompressedLength];
+        int compressedLength =
+                compressor.compress(messageData, 0, originalContentLength, compressed, 0, maxCompressedLength);
+
+        // Final compressed array
+        byte[] compressedMessage = Arrays.copyOf(compressed, compressedLength);
+
+        // Getting andes message part of the compressed message content and returning
+        return getAndesMessagePart(compressedMessage, partList.get(0).getMessageID());
+    }
+
+    /**
+     * Decompress the compressed message content, before send to subscriber
+     *
+     * @param messagePartList       Compressed message content as a collection of andes message parts
+     * @param originalContentLength Total content length of the original message content
+     * @param maxChunkSize          Maximum allowed chunk size to be stored in DB
+     * @param messageID             Message ID of the message
+     * @return Decompressed message content as a map of andes message parts and offsets
+     */
+    public Map<Integer, AndesMessagePart> getDecompressedMessage(Collection<AndesMessagePart> messagePartList, int
```

Comment:
As an improvement would it be better to create an DAO and return it instead of a Map ?